server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES93BloomFilterStoredFieldsFormat.java

```diff
@@ -0,0 +1,539 @@
+            this.bloomFilterSizeInBytes = bloomFilterSizeInBits / Byte.SIZE;
+            this.bloomFilterFieldName = bloomFilterFieldName;
+
+            this.delegateWriter = delegateWriter;
+            toClose.add(delegateWriter);
+
+            boolean success = false;
+            try {
+                bloomFilterDataOut = directory.createOutput(bloomFilterFileName(segmentInfo, segmentSuffix), context);
+                toClose.add(bloomFilterDataOut);
+                CodecUtil.writeIndexHeader(
+                    bloomFilterDataOut,
+                    STORED_FIELDS_BLOOM_FILTER_FORMAT_NAME,
+                    VERSION_CURRENT,
+                    segmentInfo.getId(),
+                    segmentSuffix
+                );
+
+                metadataOut = directory.createOutput(bloomFilterMetadataFileName(segmentInfo, segmentSuffix), context);
+                toClose.add(metadataOut);
+                CodecUtil.writeIndexHeader(
+                    metadataOut,
+                    STORED_FIELDS_BLOOM_FILTER_FORMAT_NAME,
+                    VERSION_CURRENT,
+                    segmentInfo.getId(),
+                    segmentSuffix
+                );
+
+                buffer = bigArrays.newByteArray(bloomFilterSizeInBytes, false);
+                toClose.add(buffer);
+
+                success = true;
+            } finally {
+                if (success == false) {
+                    IOUtils.closeWhileHandlingException(toClose);
+                }
+            }
+        }
+
+        @Override
+        public void startDocument() throws IOException {
+            delegateWriter.startDocument();
+        }
+
+        @Override
+        public void finishDocument() throws IOException {
+            delegateWriter.finishDocument();
+        }
+
+        @Override
+        public void writeField(FieldInfo info, int value) throws IOException {
+            delegateWriter.writeField(info, value);
+        }
+
+        @Override
+        public void writeField(FieldInfo info, long value) throws IOException {
+            delegateWriter.writeField(info, value);
+        }
+
+        @Override
+        public void writeField(FieldInfo info, float value) throws IOException {
+            delegateWriter.writeField(info, value);
+        }
+
+        @Override
+        public void writeField(FieldInfo info, double value) throws IOException {
+            delegateWriter.writeField(info, value);
+        }
+
+        @Override
+        public void writeField(FieldInfo info, StoredFieldDataInput value) throws IOException {
+            delegateWriter.writeField(info, value);
+        }
+
+        @Override
+        public void writeField(FieldInfo info, String value) throws IOException {
+            delegateWriter.writeField(info, value);
+        }
+
+        @Override
+        public void writeField(FieldInfo info, BytesRef value) throws IOException {
+            if (info.getName().equals(bloomFilterFieldName)) {
```

Comment:
I wonder if we could lookup the FieldInfo.number for the `_id` field once, and then only compare `int`s?

Reply:
Or maybe just add (bloomFilterFieldInfo != null && bloomFilterFieldInfo.number == info.number) ?

Reply:
Good point, tackled in https://github.com/elastic/elasticsearch/pull/137331/commits/a1af7b966a245fdceacabf1e723402564fc6ec1a